crates/weaver_cache/src/registry_path.rs

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The representation of a semantic convention registry path/location.
+
+use std::fmt::Display;
+use std::str::FromStr;
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+use crate::Error;
+
+/// Regex to parse a registry path supporting the following formats:
+/// - source
+/// - source@tag
+/// - source\[sub_folder\]
+/// - source@tag\[sub_folder\]
+static REGISTRY_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^(?P<source>.+?)(?:@(?P<tag>.+?))?(?:\[(?P<sub_folder>.+?)])?$")
+        .expect("Invalid regex")
+});
+
+/// Path to a semantic convention registry.
+/// The path can be a local directory or a Git URL.
+#[derive(Debug, Clone)]
+pub enum RegistryPath {
+    /// Local folder path pointing to a semantic convention registry.
+    LocalFolder {
+        /// Path to a local folder
+        path: String,
+    },
+    /// Local archive path containing a semantic convention registry.
+    LocalArchive {
+        /// Path to a local archive
+        path: String,
+        /// Sub-folder within the archive containing the semantic convention registry
+        sub_folder: Option<String>,
+    },
+    /// Remote archive containing a semantic convention registry.
+    RemoteArchive {
+        /// URL of the remote archive
+        url: String,
+        /// Sub-folder within the archive containing the semantic convention registry
+        sub_folder: Option<String>,
+    },
+    /// Git repository containing a semantic convention registry.
+    GitRepo {
+        /// URL of the Git repository
+        url: String,
+        /// Tag of the Git repository (NOT YET SUPPORTED)
```

Comment:
This should be tag, branch or commit

Reply:
Renamed `refspec` and updated the comment.